### ISSUE TITLE
Function for extracting node positions from attributes

### ISSUE THREAD

**Issue by nwlandry** [xgi-org/xgi#624]
Related to #623 and #179.

It would be great to have a function to do the following:
```python
H = xgi.ring_lattice(n, d, k, l, store_pos=True)
pos = xgi.pos_from_attributes(H, attr="pos")
xgi.draw(H, pos=pos)
```

**Comment by kaiser-dan** [comment]
Since the usual format for the `pos` argument passed to `draw` is a dictionary mapping a node id to a 2d-coordinate, could this be a more general function to extract an attribute to a dictionary? Something like
```python
H = xgi.ring_lattice(n, d, k, l, store_pos=True)
attr_dict = xgi.dict_from_attributes(H, attr="pos")
xgi.draw(H, pos=attr_dict)
```

It seems like the main advantage to a method specifying the retrieved mapping is a coordinate mapping is in implicit validation that the returned values are tuples of numbers. When this is not true, however, the `draw` function would throw an error regardless, and it seems like the intended use for this issue and the `draw` function would go hand-in-hand.

Either way, I can start on this today.

**Comment by nwlandry** [comment]
Yeah, I could see that being useful? One way to handle this is with the stats interface: `H.nodes.attrs("pos").asdict()`, but we've had lingering issues with the visibility of the stats module.
